### DISCUSSION TITLE
Popover SideBar/Body padding weirdness

### DISCUSSION THREAD

**Discussion by GiantRobots** [twilio-labs/paste#1273] [Q&A] [answered]
https://codesandbox.io/s/weird-popover-scrolling-logix-9bj7f?file=/src/TabWithRouter.tsx

If you click a Popover component some heuristic tries to keep everything in place (I'm guessing) but it causes some weird popping in the target document. 

I think the heuristic looks at if the document has a scroll bar, sets overflow to hidden (always) and if it has a scroll bar add 15px of right padding to the body. 

This heuristic results in weird popping when you click on disclosures and isn't really great if you look at the video I've linked below and watch the codesandbox link you can see it moves around. On a more complex page it can be more jostling.

https://user-images.githubusercontent.com/1903839/110695424-1a8e7f80-81a7-11eb-94db-d64df8969714.mov

**Comment by richbachman** [comment]
Hi @GiantRobots 

The Popover is currently preventing the scroll on the page body when its open. Thats causing the scroll bar to be removed and the shift. @TheSisb just discovered the bug this morning, and has a PR ready to fix: https://github.com/twilio-labs/paste/pull/1269

We'll let you know once the fix is merged.

**Comment by TheSisb** [comment, marked as answer]
I just merged the fix: https://github.com/twilio-labs/paste/pull/1269
We were accidentally blocking scrolling on the page when Popovers were open. Thanks for reporting this issue! 🙇
